cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/ZooKeeperRollerTest.java

```diff
@@ -78,6 +80,63 @@ public void testAllPodsAreRolled(VertxTestContext context)  {
                 })));
     }
 
+    @Test
+    public void testNonReadyPodsAreRestartedFirst(VertxTestContext context) {
+
+        Function<Pod, List<String>> shouldRoll = pod -> List.of("Pod was manually annotated to be rolled");
+
+        PodOperator podOperator = mock(PodOperator.class);
+        when(podOperator.isReady(any(), eq("my-cluster-zookeeper-0"))).thenReturn(true);
+        when(podOperator.isReady(any(), eq("my-cluster-zookeeper-1"))).thenReturn(false);
+        when(podOperator.isReady(any(), eq("my-cluster-zookeeper-2"))).thenReturn(true);
+        when(podOperator.listAsync(any(), any(Labels.class))).thenReturn(Future.succeededFuture(PODS));
+
+        ZookeeperLeaderFinder leaderFinder = mock(ZookeeperLeaderFinder.class);
+        when(leaderFinder.findZookeeperLeader(any(), any(), any(), any())).thenReturn(Future.succeededFuture("my-cluster-zookeeper-2"));
+
+        MockZooKeeperRoller roller = new MockZooKeeperRoller(podOperator, leaderFinder, 300_00L);
+
+        Checkpoint async = context.checkpoint();
```

Comment:
Do we need the `checkpoint` or can we just use `context.complete()` on line 107?

Reply:
Replaced the checkpoints in the test class with context.completeNow()